lib/ansible/module_utils/urls.py

```diff
@@ -870,22 +880,49 @@ def url_argument_spec():
     that will be requesting content via urllib/urllib2
     '''
     return dict(
-        url = dict(),
-        force = dict(default='no', aliases=['thirsty'], type='bool'),
-        http_agent = dict(default='ansible-httpget'),
-        use_proxy = dict(default='yes', type='bool'),
-        validate_certs = dict(default='yes', type='bool'),
-        url_username = dict(required=False),
-        url_password = dict(required=False),
-        force_basic_auth = dict(required=False, type='bool', default='no'),
+        url=dict(),
+        force=dict(default='no', aliases=['thirsty'], type='bool'),
+        http_agent=dict(default='ansible-httpget'),
+        use_proxy=dict(default='yes', type='bool'),
+        validate_certs=dict(default='yes', type='bool'),
+        url_username=dict(required=False),
+        url_password=dict(required=False),
+        force_basic_auth=dict(required=False, type='bool', default='no'),
 
     )
 
 
 def fetch_url(module, url, data=None, headers=None, method=None,
               use_proxy=True, force=False, last_mod_time=None, timeout=10):
     '''
-    Fetches a file from an HTTP/FTP server using urllib2.  Requires the module environment
+    Sends a request via HTTP(S) or FTP using urllib2 (Python2) or urllib (Python3)
+
+    Args:
+        :module (AnsibleModule): The AnsibleModule (used to get username, password etc. (s.b.).
+        :url:                    The url to use.
+        :data:                   The data to be sent (in case of POST/PUT).
+        :headers (dict):         A dict with the request headers.
+        :method (String):        "POST", "PUT", etc.
+        :use_proxy (Boolean):    Default: True
+        :force (Boolean):        If True: Do not get a cached copy (Default: False)
+        :last_mod_time:          Default: None
+        :timeout (int):          Default: 10
+
+    Returns:
+        :response: the response (body). Use `resp.body()` to read the data.
+        :info:     status code and other meta data. When a HttpError (status > 400) occurred
+                   then info['body'] contains the error response data::
+
+    Example:
+        >>> data={...}
+            resp, info = fetch_url("http://example.com", 
+                                   data=module.jsonify(data)
+                                   header={Content-type': 'application/json'},
+                                   method="POST")
+            status_code = info["body"]
+            body = resp.read()
+            if status_code >= 400 :
+                body = info['body']
```

Comment:
@abadger what of the above is not restructured text? In #16890 I had the documentation generated and it looked fine to me. The only additions compared to ".../__init__.py#L159" I can see, are:

- The documentation in this PR explicitly states the type of the parameter in some cases
- "Args:" and "Returns:" are added

The documentation at http://www.sphinx-doc.org/en/stable/domains.html#signatures defines a different format. But when I tried that, it rendered the "param" string literally.